### ISSUE TITLE
GraphSage Embedding Issue

### ISSUE THREAD

**Issue by colourful-tree** [alibaba/euler#175]
Hi, I was wondering that GraphSage SageEncoder use_id=True, which will look_up table and then get the embedding vector of input. 

My question is i don't see stop_gradient in the embedding process, so i suppose that the embedding will update with training process, but i think in the paper of GraphSage, the embedding is the input(don't update).

It's a bug?

**Comment by yangsiran** [comment]
https://github.com/williamleif/GraphSAGE/blob/master/graphsage/models.py#L230
There are also no stop_gradient in the code of author as I know. So this is used as a end-to-end trainable embedding of identity of each vertex. You may also provide pre-train embedding of vertices. In that case those embeddings may be fixed.

**Comment by colourful-tree** [comment]
yeah, i see. in the paper https://arxiv.org/pdf/1706.02216.pdf, here is the Algorithm-2:
![image](https://user-images.githubusercontent.com/17588577/68737799-81397280-061f-11ea-9ca0-de2b8f75d230.png)

xv is the input feature, if we use pre-train embedding as it, which will be fixed. otherwise, i suppose that we should use fixed embedding too(and there is hu0 <- xv in the paper, so xv should be a vector). In my thought, if we input as ID, it will auto-embedding and fixed just like embeded input. 

In the paper, author claims that model learn AGGREGATEk rather than embedding, and then naturally generalize to unseen nodes. 

Anyway, thanks a lot.
